app/models/remote_document.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class RemoteDocument
+  attr_reader :tempfile
+
+  def initialize(url)
+    @url = url
+    @tempfile = Tempfile.new.tap(&:binmode)
+  end
+
+  def download
+    open(url) { |url_file| tempfile.write(url_file.read) }
+    self
+  rescue OpenURI::HTTPError => _e
```

Comment:
is there any way to know this is a private s3 doc without a `rescue` ? This seems to work / make sense, but it's always nice to do this kind of logic without exceptions, if possible

Reply:
As far as I know, no. The only thing I can think of is to use the SDK to check - and that would mean another round-trip request. To me that would defeat the purpose.

Reply:
Nice fall back :). Do we intend to write a rake task to go through and update all the past uploaded documents to make them private again?

Reply:
I don't think we need to, @zspencer. It's a very quick and easy process to do so w/an S3 client like [Transmit]. It'd take me less than a few minutes.

[Transmit]:https://panic.com/transmit/